### PR TITLE
Net module fixes

### PR DESCRIPTION
**Pull request by dom96**

### CODE REVIEW

File: lib/pure/net.nim [hunk 1]
```diff
@@ -1432,16 +1425,22 @@ proc sendTo*(socket: Socket, address: string, port: Port, data: pointer,
...
-             data: string): int {.tags: [WriteIOEffect].} =
+             data: string) {.tags: [WriteIOEffect].} =
   ## This proc sends ``data`` to the specified ``address``,
   ## which may be an IP address or a hostname, if a hostname is specified
   ## this function will try each IP of that hostname.
   ##
+  ## If an error occurs an OSError exception will be raised.
+  ##
   ## This is the high-level version of the above ``sendTo`` function.
-  result = socket.sendTo(address, port, cstring(data), data.len, socket.domain )
```
Comment (Varriount) [inline lib/pure/net.nim]: Won't this break things?

File: lib/pure/net.nim [hunk 2]
```diff
@@ -1408,22 +1398,25 @@ proc trySend*(socket: Socket, data: string): bool {.tags: [WriteIOEffect].} =
   result = send(socket, cstring(data), data.len) == data.len
 
 proc sendTo*(socket: Socket, address: string, port: Port, data: pointer,
-             size: int, af: Domain = AF_INET, flags = 0'i32): int {.
```
Comment (Varriount) [inline lib/pure/net.nim]: Why isn't the integer returned anymore?

Comment (Araq) [reply lib/pure/net.nim]: Because exception handling is used to signal an error.

File: lib/pure/net.nim (hunk 1 above)
Comment (Araq) [reply lib/pure/net.nim]: Sure, but the changelog covers it.

**Pull request merged**